### PR TITLE
Update RKE2 hardening guide

### PR DESCRIPTION
**Pull request by macedogm**
This PR does:

1. Update RKE2 hardening guide recommendations.
2. Remove CIS v1.5 self-assessment, since it's old.
3. Fix some formatting and small typos.

### CODE REVIEW

File: docs/security/hardening_guide.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ To help ensure these above requirements are met, RKE2 can be started with the `p
 
 The following section outlines the specific actions that are taken when the `profile` flag is set to `cis-1.6`.
 
-## Host-level Requirements
+## Host-level requirements
```
[the remark this answers is not in the document]
Comment (macedogm) [reply docs/security/hardening_guide.md:25]: That's a good point. I believe that both ways are correct, it's just a matter of preferences. I'm not a doc writer expert, but I followed some of the recommendations from [SUSE Documentation Style Guide](https://documentation.suse.com/style/current/single-html/docu_styleguide/#sec-capitalization-sentence). If you believe that it's best to uppercase, then we can do it too.

[the remark this answers is not in the document]
Comment (briandowns) [reply docs/security/hardening_guide.md:25]: It adheres to the linked style guide so I guess we go with it!

File: docs/security/about_hardened_images.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-* Any binaries that are written in Go are compiled using a FIPS 140-2 compliant build process. For more information on this compiler, refer [here](https://docs.rke2.io/security/fips_support/#use-of-fips-compatible-go-compiler).
+* Any binaries that are written in Go are compiled using a FIPS 140-2 compliant build process. For more information on this compiler, refer [here](fips_support.md#use-of-fips-compatible-go-compiler).
 
 You will know if an image has been hardened as above by the image name. RKE2 publishes image lists with each release. Refer [here](https://github.com/rancher/rke2/releases/download/v1.22.3-rc1%2Brke2r1/rke2-images-all.linux-amd64.txt) for an example of a published image list.
 
-!!! note "Note:" 
-Currently, RKE2 hardened images are multi-architecture. Only the Linux AMD64 architecture is FIPS compliant. Windows and the soon-to-come s390x architectures are not FIPS compliant.
\ No newline at end of file
+!!! note
+    Currently, RKE2 hardened images are multi-architecture. Only the Linux AMD64 architecture is FIPS compliant. Windows and the soon-to-come s390x architectures are not FIPS compliant.
```
Comment (brandond) [inline docs/security/about_hardened_images.md:11]:
```suggestion
    Currently, RKE2 hardened images are multi-architecture. Only the Linux AMD64 architecture is FIPS compliant. Windows and s390x are not FIPS compliant.
```

**Review by brandond — CHANGES_REQUESTED** [review]
One nit, LGTM otherwise!

**Review by brandond — APPROVED** [review]

**Comment by macedogm** [comment]
This PR is ready, but I'm moving to draft in order to hold it's merge, so QA can review it for Rancher 2.6.5 release. CCing @anupama2501 .

**Comment by macedogm** [comment]
PR is ready to merge, as the review got a green light in https://github.com/rancher/rancher/issues/37242#event-6493743200.

@brandond do we need an additional reviewer/approval?

**Comment by macedogm** [comment]
@brandond I fixed the spelling error, but the pipeline now failed due to another issue. Doesn't seem to be related to the docs update.

**Comment by macedogm** [comment]
@brandond pipeline is green now. Can we merge it, please, or do we need another approval?

**Review by jossemargt-3pillar — APPROVED** [review]

**Comment by macedogm** [comment]
@brandond is the code freeze already lifted? If yes, can we get this PR merged?

**Review by dereknola — APPROVED** [review]

**Pull request merged**